tests/optimagic/optimization/test_error_penalty.py

```diff
@@ -54,9 +68,11 @@ def test_penalty_derivatives(func, deriv):
     partialed = functools.partial(
         func, constant=constant, slope=slope, x0=x0, dim_out=dim_out
     )
-    expected = first_derivative(partialed, x)
+    expected = first_derivative(
+        partialed, x, unpacker=lambda x: x.internal_value(solver_type)
```

Comment:
Maybe we should also allow `unpacker` to be an `AggregationLevel`, in which case we set the unpacker to `lambda x: x.internal_value(unpacker)`?

Reply:
This is mostly used internally (for example not documented in any how-to guide or tutorial) and I am hesitant to add more logic to the derivatives before the re-write.